pkg/controllers/util/federatedclient/client.go

```diff
@@ -311,6 +315,7 @@ func (f *federatedClientFactory) processQueueItem(ctx context.Context) {
 	f.kubeInformerCache[name] = kubeInformerFactory
 	f.dynamicInformerCache[name] = dynamicInformerFactory
 	f.mu.Unlock()
+	f.addPodInformer(kubeInformerFactory, kubeClientset)
```

Comment:
this function should be called within the critical section above (i.e. before `f.mu.Unlock()` on line 317).  if not there may be a chance that the shared pod informer is accessed by another goroutine before this function is called, causing the default pod informer to be constructed instead